Remove Enrich and Filter for ASP.NET Core metrics (#4981)
open-telemetry/opentelemetry-dotnet: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md

to `http` or `http/dup`.
([#4934](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4934))

* **Breaking**: Removed `Enrich` and `Filter` support for **metrics**

>   instrumentation. The `Enrich` API for `http.server.request.duration` will be

Reviewer: ```suggestion
  instrumentation. An `Enrich` API for `http.server.request.duration` will be
```

  available for users targeting `.NET8.0` or newer frameworks. For details see
  [Enrich the ASP.NET Core request
  metric](https://learn.microsoft.com/aspnet/core/log-mon/metrics/metrics?view=aspnetcore-8.0#enrich-the-aspnet-core-request-metric).

>   `Filter` capability will not be available for any of the targeted .NET

Reviewer: ```suggestion
  `Filter` capability is not available for any .NET
```

  versions.
  ([#4981](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4981))

> 

Reviewer: consider opening a dedicated issue for users to present the case of Filter in metrics. Include a link to that from here, so folks affecting with this change can request for bringing it back/explain the use case etc.

Reviewer: https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/CHANGELOG.md#changed You may borrow some ideas from this changelog for breaking change/feedback-issue, if you like that wording!

Author: Thanks!. updated.

to `http` or `http/dup`.
([#4934](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4934))

> * **Breaking**: Removed `Enrich` and `Filter` support for **metrics**

Reviewer: I think its best to explicitly mention `AspNetCoreMetricsInstrumentationOptions` is no longer available as the only reason for its existence (enrich/filter) were removed.

to `http` or `http/dup`.
([#4934](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4934))

> * **Breaking**: Removed `Enrich` and `Filter` support for **metrics**
>   instrumentation. With this change, `AspNetCoreMetricsInstrumentationOptions`
>   is no longer available. An `Enrich` API for `http.server.request.duration`
>   will be available for users targeting `.NET8.0` or newer frameworks. For
>   details see [Enrich the ASP.NET Core request
>   metric](https://learn.microsoft.com/aspnet/core/log-mon/metrics/metrics?view=aspnetcore-8.0#enrich-the-aspnet-core-request-metric)
> 
>   `Filter` capability is not available for any .NET versions. Please [share your
>   feedback
>   here](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4982) if
>   you are affected.
>   ([#4981](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4981))

Reviewer 2: ```suggestion
* **Breaking**: Removed `Enrich` and `Filter` support for **metrics**
  instrumentation. With this change, `AspNetCoreMetricsInstrumentationOptions`
  is no longer available.
  ([#4981](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4981))

  * `Enrich` migration:

    An enrichment API for the `http.server.request.duration` metric is available
    inside AspNetCore for users targeting .NET 8.0 (or newer). For details see:
    [Enrich the ASP.NET Core request
    metric](https://learn.microsoft.com/aspnet/core/log-mon/metrics/metrics?view=aspnetcore-8.0#enrich-the-aspnet-core-request-metric).
  
  * `Filter` migration:

    There is no comparable filter mechanism currently available for any .NET
    version. Please [share your
    feedback](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4982)
    if you are impacted by this feature gap.

    > **Note**
    > The [View
    API](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/metrics/customizing-the-sdk#select-specific-tags)
    may be used to drop dimensions.
```
